### PR TITLE
add finished task 11

### PR DESCRIPTION
**Pull request by tjhind**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -86,15 +86,33 @@ describe("/api/articles", () => {
...
+    return request(app)
+      .get("/api/articles?topic=7777")
       .expect(404)
       .then((response) => {
-        expect(response.body.msg).toBe("Invalid path");
+        expect(response.body.msg).toBe("No articles found");
       });
   });
 });
+test("GET 404: responds with an error when wrong endpoint is requested", () => {
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:108]: This bad path test isn't related to the current test and is actually covering all bad paths so should be in its own describe block, or if we already have a similar test then this one won't be needed. If a test auto passes it's a good indicator that we may have written a bad test.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -86,15 +86,33 @@ describe("/api/articles", () => {
...
+      });
+  });
+  test("GET 404: responds with an error if no articles are found for topic query value", () => {
+    return request(app)
+      .get("/api/articles?topic=7777")
       .expect(404)
       .then((response) => {
-        expect(response.body.msg).toBe("Invalid path");
+        expect(response.body.msg).toBe("No articles found");
       });
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:105]: Nice work on testing so far, could we also consider what would happen if we queried a valid topic that happens to have no articles associated with it? This should be a 200 but an empty response, not an error.

File: __tests__/app.test.js (hunk 1 above)
Comment (tjhind) [reply __tests__/app.test.js:108]: Have removed this test as there is already an appropriate test block for it

File: __tests__/app.test.js (hunk 2 above)
Comment (tjhind) [reply __tests__/app.test.js:105]: Added advanced error handling for this task

**Pull request merged**